File: Leetcode/Python/reverseInteger.py

#Problem link : https://leetcode.com/problems/reverse-integer/
def reverse(x):
    s = str(abs(x))
    ans = int (s[::-1])
    if (ans > (pow(2,31)-1) or ans < pow(2,-31)):
        return 0
    elif (x >= 0):
        return ans
    return ans - 2*ans




num = int(input())
print(reverse(num))

#Example test case
# Input : 123
# Output : 321
# Input :  - 123
# Output : -321

File: Interviewbit/Math/Reverse_integer.cpp
// https://www.interviewbit.com/problems/reverse-integer/
// O(logN)

int Solution::reverse(int A) {
    
    long long int B = 0;
    int sign = A < 0 ? 1 : 0;
    A = abs(A);
    
    while( A ){
        B = B*10+A%10;
        A = A/10;
        
        if( B > INT_MAX )
            return 0;
    }
    if( B > INT_MAX)
        return 0;
    
    if( sign )
        B = -B;
    
    if( B >= INT_MIN)
        return B;
        
    return 0;
}
